### DISCUSSION TITLE
Robot stucks into the plane.urdf

### DISCUSSION THREAD

**Discussion by KH-HSU** [bulletphysics/bullet3#3389] [General]
Happy New Year @erwincoumans and everyone

Problem:
When the robot walked from the board, sometimes the robot will stuck into the plane which are shown in pictures below. This problem also happened when I used Heightfield.

If the robot just walked on the plane without the red board in the pictures, the robot will sink a little bit but will not stuck into the plane. ( The red line is for RL experiment without collision, so it will not effect collision.)
 
![sink1](https://user-images.githubusercontent.com/25145866/103497217-d2d43a80-4e7b-11eb-9ae5-ca6c93b4f033.PNG)
![sink2](https://user-images.githubusercontent.com/25145866/103497221-d49dfe00-4e7b-11eb-867b-98bb62f738b3.PNG)

What I have done
1. set timestep to 0.001
2. set numSolverIterations to 1000
3. Use the smallest resolution to generate mesh of the robot in Solidworks (also tried in STL and obj format)

but the problems still exists
Does anyone have met this problem before? Please give me some suggestions, thank you!

**Comment by KH-HSU** [comment]
I solved it by 
1. putting the board base under the plane ( I put it just right on the plane before ), 
2. set the numSolverIterations to 1000. 
I'm still curious if there are other solutions because I want to use Heightfield to do experiment.

**Comment by BluePrintRandom** [comment]
a plane has zero thickness - try a very short cube instead using the cube
bound.

**Comment by KH-HSU** [comment]
@BluePrintRandom 
I'm a bit confused because the plane.urdf given by Pybullet has thickness which can be found in 
[plane.urdf](https://github.com/bulletphysics/bullet3/blob/master/examples/pybullet/gym/pybullet_data/plane.urdf#L22), and I made the red cube by loading [cube.obj](https://github.com/bulletphysics/bullet3/blob/master/examples/pybullet/gym/pybullet_data/cube.obj) into urdf

**Comment by BluePrintRandom** [comment]
you can use CCD to ensure that a object does not pop through another - but
it's typically easier to simulate a more realistic environment (ie a
thicker floor)

**Comment by KH-HSU** [comment]
@BluePrintRandom thank you, I will try CCD.
But the problem still exists when the floor thickness is 4 cm.

I put the green cube under the plane and fix it to solve the problem which is shown in the picture below![below_plane](https://user-images.githubusercontent.com/25145866/103609691-5e6fc900-4f59-11eb-9e01-3a8abc8346f2.PNG) 
the red line without collision property is just right on the plane.

I wonder if I just put the floor right on the plane, there might be some small space that will make the robot stuck in the mesh.
